keras/src/layers/preprocessing/image_preprocessing/random_apply.py

```diff
@@ -0,0 +1,151 @@
+            for the entire batch. If `False`, it is made independently for each
+            input. Defaults to `False`.
+        seed: Optional integer to ensure reproducibility.
+
+    Inputs: A tensor (rank 3 for single input, rank 4 for batch input). The
+        input can have any dtype and range.
+
+    Output: A tensor with the same shape and dtype as the input, with the
+        transformation layer applied to selected inputs.
+    """
+
+    def __init__(
+        self,
+        layer,
+        rate=0.5,
+        batchwise=False,
+        seed=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        if not (0 <= rate <= 1.0):
+            raise ValueError(
+                f"rate must be in range [0, 1]. Received rate: {rate}"
+            )
+        self._layer = layer
+        self._rate = rate
+        self.batchwise = batchwise
+        self.seed = seed
+        self.generator = SeedGenerator(seed)
+        self.built = True
+
+    def get_random_transformation(self, data, training=True, seed=None):
+        if not training:
+            return None
+
+        if seed is None:
+            seed = self._get_seed_generator(self.backend._backend)
+
+        if isinstance(data, dict):
+            inputs = data["images"]
+        else:
+            inputs = data
+
+        input_shape = self.backend.shape(inputs)
+        if self.batchwise:
+            should_augment = self._rate > self.backend.random.uniform(
+                shape=(), seed=seed
+            )
+        else:
+            batch_size = input_shape[0]
+            random_values = self.backend.random.uniform(
+                shape=(batch_size,), seed=seed
+            )
+            should_augment = random_values < self._rate
+
+            ndims = len(input_shape)
+            ones = [1] * (ndims - 1)
+            broadcast_shape = tuple([batch_size] + ones)
+            should_augment = self.backend.numpy.reshape(
+                should_augment, broadcast_shape
+            )
+
+        return {
+            "should_augment": should_augment,
+            "input_shape": input_shape,
+        }
+
+    def transform_images(self, images, transformation, training=True):
+        if not training or transformation is None:
+            return images
+
+        should_augment = transformation["should_augment"]
+
+        if hasattr(self._layer, "get_random_transformation"):
+            layer_transform = self._layer.get_random_transformation(
+                images, training=True
+            )
+            augmented = self._layer.transform_images(
+                images, layer_transform, training=True
+            )
+        else:
+            augmented = self._layer(images)
+
+        return self.backend.numpy.where(should_augment, augmented, images)
+
+    def call(self, inputs):
+        if isinstance(inputs, dict):
+            return {
+                key: self._call_single(input_tensor)
```

Comment:
Seems incorrect -- it should be something like:

```
if apply:
   return self._layer(inputs)
```

This is independent of whether inputs are a dict or not